myhpi/feed/middleware.py

```diff
@@ -0,0 +1,26 @@
+from django.http import HttpResponseRedirect
+
+from myhpi import settings
+from myhpi.feed.models import NewsFeed
+
+
+class FeedRedirectMiddleware:
```

Comment:
you could also implement this inside the serve() of RootPage, that would only do the check when it is actually necessary and not with every request

Reply:
But that would increase coupling. Currently disabling/removing the app is very easy, it is never mentioned / used in core.

Reply:
In my opinion, it would be okay to introduce some change to core, as myHPI (this application) is developed specifically for our myHPI website and not meant as a generic CMS. Hence, we don't need to be able to deactivate/remove the newsfeed app per installation and that small amount of coupling would be acceptable.

However, I'm not that familiar with Django patterns, so I can't really argue on whether there is even a need to replace this middleware or not.

Reply:
@lukasrad02 Felix and I had a conversation about the conflict of cool new features vs. size of codebase/maintainability. I made the argument that as long as fancy features are loosely coupled, future maintainer could easily remove them without the need to understand/update/fix them and therefore delay or block the maintenance of the core app.

The middleware is probably fine, as it doesn't do much. I just thought about it because we always struggle with pageload times and this is code that is not necessary on all pages but the root page.

Another way that we could pursue this is by adding some kind of "plugin" mechanism. Again citing ephios: we are sending out custom django signals to "plugins" (other django apps within the same repo or installed separately) and let them inject stuff at various places. e.g. there is a signal that says "hey plugin, give me some html to append to the homepage if you want to". This way, the additional code is only run when it is necessary.

Reply:
I'm fine with you two deciding on the best approach to do this. Especially when it comes to performance or plugins in Django, I would not be able to give any qualified arguments.